compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SuspendLambdaLowering.kt

```diff
@@ -209,19 +208,43 @@ private class SuspendLambdaLowering(context: JvmBackendContext) : SuspendLowerin
                     it.valueParameters[0].type.isKotlinResult()
         }
         return addFunctionOverride(superMethod, irFunction.startOffset, irFunction.endOffset).apply {
-            body = irFunction.moveBodyTo(this, mapOf())?.transform(object : IrElementTransformerVoid() {
-                override fun visitGetValue(expression: IrGetValue): IrExpression {
-                    val parameter = (expression.symbol.owner as? IrValueParameter)?.takeIf { it.parent == irFunction }
-                        ?: return expression
-                    val field = fields[parameter.index + if (irFunction.extensionReceiverParameter != null) 1 else 0]
-                    val receiver = IrGetValueImpl(UNDEFINED_OFFSET, UNDEFINED_OFFSET, dispatchReceiverParameter!!.symbol)
-                    return IrGetFieldImpl(expression.startOffset, expression.endOffset, field.symbol, field.type, receiver)
-                }
-            }, null)
+            val localVals: List<IrVariable> = fields.mapIndexed { index, field ->
+                val varDescriptor = WrappedVariableDescriptor()
```

Comment:
Maybe refactor this variable creation to `buildVariable`?

Reply:
Yup, happy to! Should I move it to that irFactory in the context?

Reply:
I mean use [this existing function](https://github.com/JetBrains/kotlin/blob/5aaaa3881dfea4b58473dc3572be6f9ad220ea2c/compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt#L322) here, mainly to get rid of manual descriptor creation and binding

Reply:
Perfect, thanks! Also caught an "optimized import" in an unrelated file.